src/d3a/models/strategy/pv.py

```diff
@@ -104,6 +102,10 @@ def event_activate(self):
         if self.use_market_maker_rate:
             self.area_reconfigure_event(initial_selling_rate=GlobalConfig.market_maker_rate,
                                         validate=False)
+        for time_slot in generate_market_slot_list():
```

Comment:
Same here, should be refactored to its own function and called from the `area_reconfigure_event` method too.